docs/examples/hairpin_fitting/hairpin_unfolding.rst

```diff
@@ -0,0 +1,339 @@
+`fit["dna_handles/Lc"].fixed = True`. For this particular data set, the fit converges without fixing parameters. 
+We next set the initial guesses for the unfolded hairpin::
+
+    fit["dna_ss_hairpin/Lp"].lower_bound = 0.5  # in nanometers
+    fit["dna_ss_hairpin/Lp"].value = 1.5  # in nanometers
+    fit["dna_ss_hairpin/Lp"].upper_bound = 2.0  # in nanometers
+
+    fit["dna_ss_hairpin/Lc"].value = 0.02  # in microns
+    fit["dna_ss_hairpin/Lc"].lower_bound = 0.001  # in microns
+
+    fit["dna_ss_hairpin/St"].value = 500  # in pN
+    fit["dna_ss_hairpin/St"].upper_bound = 2000  # in pN
+
+Fit all the data and plot the result::
+
+   >>> fit.fit()
+   >>> print(fit.params)
+   Name                  Value       Unit      Fitted   Lower bound    Upper bound
+   --------------      ------------ --------  -------  -------------  -------------
+   dna_handles/Lp        34.7309    [nm]        True    30              100
+   dna_handles/Lc         1.75753   [micron]    True     0.00034        inf
+   dna_handles/St       904.301     [pN]        True     1              inf
+   kT                     4.11      [pN*nm]     False    3.77           8
+   dna_ss_hairpin/Lp      0.785985  [nm]        True     0.5            2
+   dna_ss_hairpin/Lc      0.0215748 [micron]    True     0.001          inf
+   dna_ss_hairpin/St   2000         [pN]        True     1              2000
+
+As can be seen from the table, most fitted parameters converge and have values in the expected range. However, the stretch modulus of the hairpin hits the upper bound of 2000 pN, indicating that this parameter did not converge.
+When observing that a parameter does not converge, it is important to go back to the fit and see how it can be improved. In this case, increasing the upper bound for `dna_ss_hairpin/St` does not visually change the fit and does not result in convergence;
+the stretch modulus of the DNA handles and the hairpin cannot be optimized at the same time. A solution would be to use the freely jointed chain, instead of the extensible freely jointed chain to fit the unfolded hairpin, which is equivalent to setting
+a very large value for `dna_ss_hairpin/St`::
+
+    fit["dna_ss_hairpin/St"].upper_bound = 100000 
+    fit["dna_ss_hairpin/St"].value = 100000  # in pN
+    fit["dna_ss_hairpin/St"].fixed = True 
+    fit.fit()
+
+Plot the result::
+
+    plt.figure()
+    fit[dna_handles_force].plot()
+    fit[dna_handles_and_hairpin_force].plot()
+    plt.xlabel(r"Distance ($\mu$m)")
+    plt.ylabel("Force (pN)")
+    Lc_hairpin = fit["dna_ss_hairpin/Lc"].value*1000
+    plt.title(f"Fitted hairpin length is {Lc_hairpin:0.1f} nm")
+
+.. image:: fit_all.png
+
+The expected contour length for the hairpin was ~17 nm and the fitted length is 21.7 nm. 
+
+The next step is to study the confidence intervals and quality of the fit, for example using the :ref:`likelihood profile <ple_confidence_intervals>`.
```

Comment:
It's interesting that if I make a likelihood profile, 17 isn't in the CI.

What is quite unfortunate in these types of experiments, is that the estimate for Lc depends quite strongly on any distance offset present. You can see this if you profile with an offset present (not suggesting we should add one).

Reply:
Good point. My experience is that Lc cannot be optimized in combination with a distance offset. Fortunately, it is usually only the unfolding length that is unknown. So, Lc could be fixed during the fit.

Reply:
Not sure if this example is the right place to mention how to handle different bead sizes. But this should be addressed somewhere.. I'll make a note to myself about it. A data set with unequal bead sizes would actually have been better for this tutorial...

Reply:
I like that you added this section.